### PR TITLE
Python sorting algorithm

### PR DESCRIPTION
**Pull request by Boy-timi**

### CODE REVIEW

File: codes/python/chapter_sorting/insertion_sort.py [hunk 1]
```diff
@@ -8,3 +8,19 @@
 sys.path.append(osp.dirname(osp.dirname(osp.abspath(__file__))))
 from include import *
 
+"直接插入排序"
```
Comment (krahets) [inline codes/python/chapter_sorting/insertion_sort.py:11]: Please check the function name, ensuring it matches the Java version, which should be `""" 插入排序 """`.

File: codes/python/chapter_sorting/merge_sort.py [hunk 2]
```diff
@@ -8,3 +8,41 @@
 sys.path.append(osp.dirname(osp.dirname(osp.abspath(__file__))))
 from include import *
 
+
+"""
+另一种 思路实现归并排序
```
Comment (krahets) [inline codes/python/chapter_sorting/merge_sort.py:13]: Thanks for the hard work!
Considering the correspondence, it is not recommended to create new codes out of the Java version.
Therefore, you'd better remove this function.

File: codes/python/chapter_sorting/quick_sort.py [hunk 3]
```diff
@@ -8,3 +8,24 @@
 sys.path.append(osp.dirname(osp.dirname(osp.abspath(__file__))))
 from include import *
 
+
+"另一种思维 实现快速排序"
```
Comment (krahets) [inline codes/python/chapter_sorting/quick_sort.py:12]: Same as above

File: codes/python/chapter_sorting/insertion_sort.py (hunk 1 above)
Comment (krahets) [reply codes/python/chapter_sorting/insertion_sort.py:11]: And the function name should be `insertion_sort()`.

File: codes/python/chapter_sorting/bubble_sort.py [hunk 4]
```diff
@@ -8,3 +8,30 @@
...
+def bubble_sort(nums):
+    n=len(nums)
+    # 外循环：待排序元素数量为 n-1, n-2, ..., 1
+    for i in range(n-1,-1,-1):
+        # 内循环：冒泡操作
+        for j in range(i):
+            if nums[j]>nums[j+1]:
+                nums[j],nums[j+1]=nums[j+1],nums[j]
+"冒泡排序（标志优化）"
+def bubbleSortWithFlag(nums):
```
Comment (krahets) [inline codes/python/chapter_sorting/bubble_sort.py:21]: It shouble be `def bubble_sort_with_flag(nums):`

**Review by krahets — CHANGES_REQUESTED** [review]
Thanks, @Boy-timi! Some comments were added and need to be addressed.

**Comment by Boy-timi** [comment]
谢谢。第一次使用，我尝试修改一下



| |
***@***.***
|
|
***@***.***
|




---- Replied Message ----
| From | Yudong ***@***.***> |
| Date | 11/25/2022 17:12 |
| To | ***@***.***> |
| Cc | ***@***.***>***@***.***> |
| Subject | Re: [krahets/hello-algo] Python sorting algorithm (PR #17) |

@krahets requested changes on this pull request.

Thanks, @Boy-timi! Some comments were added and need to be addressed.

In codes/python/chapter_sorting/bubble_sort.py:

> @@ -8,3 +8,30 @@

 sys.path.append(osp.dirname(osp.dirname(osp.abspath(__file__))))



Please replace the content of the file header with yours.
(Don't forget to replace file headers of other files that you created)
Thanks.

'''File: bubble_sort.pyCreated Time: 2022-11-25Author: YOUR-NAME (YOUR-EMAIL)'''

In codes/python/chapter_sorting/bubble_sort.py:

> @@ -8,3 +8,30 @@

 sys.path.append(osp.dirname(osp.dirname(osp.abspath(__file__))))

 from include import *

 

+"冒泡排序"



All the code should be formatted by the standard tool for better readability.

Shortcuts:
Ctrl + A -> Ctrl + K -> Ctrl + F for VSCode
Ctrl + Alt + L for IDEA PyCharm.

In codes/python/chapter_sorting/bubble_sort.py:

> @@ -8,3 +8,30 @@

 sys.path.append(osp.dirname(osp.dirname(osp.abspath(__file__))))

 from include import *

 

+"冒泡排序"

+def bubble_sort(nums):

+    n=len(nums)

+    # 外循环：待排序元素数量为 n-1, n-2, ..., 1

+    for i in range(n-1,-1,-1):

+        # 内循环：冒泡操作

+        for j in range(i):

+            if nums[j]>nums[j+1]:

+                nums[j],nums[j+1]=nums[j+1],nums[j]

+"冒泡排序（标志优化）"



We use """ 冒泡排序（标志优化） """ to comment on the function.

In codes/python/chapter_sorting/insertion_sort.py:

> @@ -8,3 +8,19 @@

 sys.path.append(osp.dirname(osp.dirname(osp.abspath(__file__))))

 from include import *

 

+"直接插入排序"



Please check the function name, ensuring it matches the Java version, which should be """ 插入排序 """.

In codes/python/chapter_sorting/merge_sort.py:

> @@ -8,3 +8,41 @@

 sys.path.append(osp.dirname(osp.dirname(osp.abspath(__file__))))

 from include import *

 

+

+"""

+另一种 思路实现归并排序



Thanks for the hard work!
Considering the correspondence, it is not recommended to create new codes out of the Java version.
Therefore, you'd better remove this function.

In codes/python/chapter_sorting/quick_sort.py:

> @@ -8,3 +8,24 @@

 sys.path.append(osp.dirname(osp.dirname(osp.abspath(__file__))))

 from include import *

 

+

+"另一种思维 实现快速排序"



Same as above

In codes/python/chapter_sorting/bubble_sort.py:

> @@ -8,3 +8,30 @@

 sys.path.append(osp.dirname(osp.dirname(osp.abspath(__file__))))

 from include import *

 

+"冒泡排序"

+def bubble_sort(nums):

+    n=len(nums)

+    # 外循环：待排序元素数量为 n-1, n-2, ..., 1

+    for i in range(n-1,-1,-1):

+        # 内循环：冒泡操作

+        for j in range(i):

+            if nums[j]>nums[j+1]:

+                nums[j],nums[j+1]=nums[j+1],nums[j]

+"冒泡排序（标志优化）"

+def bubbleSortWithFlag(nums):



It shouble be def bubble_sort_with_flag(nums):

In codes/python/chapter_sorting/insertion_sort.py:

> @@ -8,3 +8,19 @@

 sys.path.append(osp.dirname(osp.dirname(osp.abspath(__file__))))

 from include import *

 

+"直接插入排序"



And the function name should be insertion_sort().

**Comment by krahets** [comment]
@Boy-timi 好的，建议看一下 `codes/python/chapter_array_and_linkedlist/array.py` 的代码格式，尽量保持统一哈～

**Pull request closed without merging**